neo4j-store/src/main/kotlin/org/ostelco/prime/storage/graph/Neo4jStore.kt

```diff
@@ -488,6 +489,26 @@ object Neo4jStoreSingleton : GraphStore {
                             ifTrue = {},
                             ifFalse = { NotFoundError(type = identityEntity.name, id = identity.id) })
                 }.bind()
+
+                /* If removal of payment profile fails, then the customer will be deleted
+                   in neo4j but will still be present in payment backend. In that case the
+                   profile must be removed from the payment backend manually. */
+                paymentProcessor.removePaymentProfile(customerId)
+                        .map {
+                            Unit
+                        }.flatMapLeft {
+                            if (it is org.ostelco.prime.paymentprocessor.core.NotFoundError) {
+                                /* Ignore. Customer has not bought products yet. */
+                                Unit.right()
+                            } else {
+                                logger.error(NOTIFY_OPS_MARKER,
+                                        "Removing corresponding payment profile when removing customer $customerId " +
+                                                "failed with error ${it.message} : ${it.description}")
+                                PartiallyNotDeletedError(type = customerEntity.name,
+                                        id = "Failed to remove corresponding payment profile when removing customer $customerId",
+                                        error = it).left()
+                            }
+                        }.bind()
             }.fix()
         }.unsafeRunSync()
                 .ifFailedThenRollback(transaction)
```

Comment:
I do not think we need new Error since we do not delete partially.  If it fails to delete in Stripe, it will rollback for Neo4j too.